doc/arch/adr-002-rest-api.md

```diff
@@ -0,0 +1,74 @@
+| ------------------------------------------ | ------ | ------------------------ | -------------- | ---------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------ |
+| `/boundaries/`                             | GET    | -                        | Logged In User | 200 OK                                                     | Returns list of all boundaries the user has access to                                                              |
+| `/boundaries/?utilities=1,3,5,8`           | GET    | -                        | Logged In User | 200 OK,<br />401 Not Authorized                            | Returns list of boundaries in the given utilities. If the user does not have access to that utility, returns a 401 |
+| `/boundaries/`                             | POST   | `{ ... }`                | Contributor    | 201 Created,<br />400 Bad Request,<br />401 Not Authorized | Creates a new boundary if the payload is correct and the user is authorized                                        |
+| `/boundaries/{id}/`                        | GET    | -                        | Logged In User | 200 OK,<br />404 Not Found                                 | Returns boundary details if the users has access to it, else 404s. Includes all details about the boundary.        |
+| `/boundaries/{id}/shape/`                  | PUT    | `{ ... }`                | Contributor    | 200 OK,<br />404 Not Found                                 | Updates a boundary's shape                                                                                         |
+| `/boundaries/{id}/reference-images/`       | POST   | `{ ... }`                | Contributor    | 200 OK,<br />404 Not Found                                 | Adds a new reference image to the boundary                                                                         |
+| `/boundaries/{id}/submit/`                 | POST   | `{ notes }`              | Contributor    | 200 OK,<br />404 Not Found                                 | Submits the boundary                                                                                               |
+| `/boundaries/{id}/review/`                 | POST   | `{ notes, annotations }` | Validator      | 200 OK,<br />404 Not Found                                 | Reviews a boundary                                                                                                 |
+| `/boundaries/{id}/review/annotations/{id}` | PUT    | `{ annotation }`         | Validator      | 200 OK,<br />404 Not Found                                 | Updated an annotation                                                                                              |
```

Comment:
This is one of those things that I'm a little conflicted about. Given that there's multiple Reviews in the database, we _should_ have it in the endpoint to be explicit.

However, we do not allow editing old Reviews, only the current ones. Since there is only ever one active Review, we can figure it out in the back-end, and make it transparent to the front-end.

Does that make sense?

Reply:
That was my understanding also. Worth a quick note in this doc, or no?

Reply:
[d4675f2](https://github.com/azavea/iow-boundary-tool/pull/115/commits/d4675f268bbb03ed74d80efd7e4be8e457daf2d7)